sharding/indexHashedNodesCoordinator.go

```diff
@@ -637,12 +648,14 @@ func (ihgs *indexHashedNodesCoordinator) computeNodesConfigFromList(
 		sort.Sort(validatorList(leavingList))
 	}
 
+	nbShards := core.MaxInt(0, len(eligibleMap)-1)
+
```

Comment:
Shouldn't error when nbShards==0?

Reply:
you could still have enough in the waiting lists to fill all shards, maybe... but there is a problem with the state in the trie if there are no validators in the eligible map. I added some time ago a "// TODO: compare with previous nodesConfig if exists" in order to validate that what comes from the startOfEpochBlock matches what the validator had in the previous epoch. If the epochs are consecutive, there is a problem and maybe an error should occur.